test/common.js

```diff
@@ -616,3 +616,18 @@ exports.WPT = {
     assert.fail(undefined, undefined, `Reached unreachable code: ${desc}`);
   }
 };
+
+// Useful for testing expected internal/error objects
+exports.expectsError = function expectsError(code, type, message) {
+  return function(error) {
+    assert.strictEqual(error.code, code);
+    if (type !== undefined)
+      assert(error instanceof type, 'error is not the expected type');
+    if (message !== undefined) {
+      if (!util.isRegExp(message))
+        message = new RegExp(String(message));
```

Comment:
I'd prefer that if it's a string, match for equality. This way, we don't get weird results due to regexp special characters in strings:

```js
const re = new RegExp(String('foo + bar'));
re.test('foo + bar'); // false!
```